sway/desktop/render.c

```diff
@@ -103,18 +98,17 @@ static void set_scale_filter(struct wlr_output *wlr_output,
 }
 
 static void render_texture(struct render_context *ctx, struct wlr_texture *texture,
-		const struct wlr_fbox *src_box, const struct wlr_box *dst_box,
+		const struct wlr_fbox *_src_box, const struct wlr_box *dst_box,
 		enum wl_output_transform transform, float alpha) {
-	struct wlr_renderer *renderer = ctx->renderer;
 	struct sway_output *output = ctx->output;
 
 	struct wlr_box proj_box = *dst_box;
 	scale_box(&proj_box, output->wlr_output->scale);
 
-	float matrix[9];
-	enum wl_output_transform inv_transform = wlr_output_transform_invert(transform);
-	wlr_matrix_project_box(matrix, &proj_box, inv_transform, 0,
-		output->wlr_output->transform_matrix);
+	struct wlr_fbox src_box = {0};
+	if (_src_box) {
+		src_box = *_src_box;
+	}
 
 	pixman_region32_t damage;
 	pixman_region32_init(&damage);
```

Comment:
Maybe we need to use the scaled box here?

Reply:
Turns out the dst_box shouldn't be scaled to fix the issue. Callers are already responsible for scaling which seems odd to me. I'll leave it to try to reduce risk of regressions till all this gets blown away by scene.